### PR TITLE
rsx: Texturing improvements

### PR DESCRIPTION
**Pull request by kd-11**
- Remember to update format class when doing mipmap gather in case of cross-aspect respec.
- Fix handling of 16-bit formats in blit engine. The logical representation of RGB565 is very different than R16U when interpolation is enabled.

Fixes https://github.com/RPCS3/rpcs3/issues/10875
Fixes https://github.com/RPCS3/rpcs3/issues/9076

### CODE REVIEW

**Comment by Crafty-The-Fox** [comment]
This fixes the "wireframe like" issue. but I'm not sure if this was hiding behind it. Parts of the models start to disappear when the game goes farther and farther from the camera. In the 1st screenshot some of the crowd is red instead of the assumed green, and textures next to the deck are scrambled. Those issues are minor though.
image

In the second screenshot you can also see some clipping with the models happening
image

In the 3rd screenshot, it looks normal. This is probably because the camera is closer to the crowd.
image



I compared the 2nd image to the screenshots from console that @shockdude posted in the issue, so it seems to be a RPCS3 issue.

**Comment by shockdude** [comment]
I can replicate @Crafty-The-Fox's first image.
In the lower-right, in front of that funny structure, a woman's head is invisible despite being visible in master. Also the dancers in the PR are red when they should be green.

**PR 11163**
Log: [RPCS3_pr11163.log.gz](https://github.com/RPCS3/rpcs3/files/7551799/RPCS3_pr11163.log.gz)
RSX Capture: https://drive.google.com/file/d/1JPl71-AzL9I7djGnGXKJOZnwJ1_JMlOt/view?usp=sharing
BLUS30526_screenshot_2021_11_16_pr11163

**Master 0.0.19-13024**
Log: [RPCS3_master13024.log.gz](https://github.com/RPCS3/rpcs3/files/7551817/RPCS3_master13024.log.gz)
RSX Capture: https://drive.google.com/file/d/1-0-coPzVp68qJNqHLQleR14cfZAz2b2G/view?usp=sharing
BLUS30526_screenshot_2021_11_16_master13024

**Console**
DJ Hero 2 bujaclub

**Comment by kd-11** [comment]
Format mismatch should be fixed. However, I had to touch blit engine, which means any number of games may have been broken by the changes. Freezing until more general testing is done.

**Comment by Crafty-The-Fox** [comment]
Good news, the new build seems to fix the issue.

2nd, is that ive tested Skate 3 and LBP:K, and they seem to play fine. Il test a couple more games that I have on hand.

**Comment by shockdude** [comment]
New build looks great for me too.
BLUS30526_screenshot_2021_11_17_pr11163

Also tested DJ Hero 1 and that has fixed crowds as well.

**Comment by Linear524** [comment]
I would like to test this PR but I can't build an executable on my current PC... May I ask compiled artifact of this PR in order to test it on my PS3 titles collection ?

**Comment by MSuih** [comment]
You can download artifacts from Cirrus, either by using the bot in our Discord server or by following these steps:

1. Scroll to bottom of the page for the sercion about checks, click on "Show all details" if it isn't visible
2. Find the task for Windows and click on Details.
3. Scroll to bottom and click "View more details on Cirrus CI"
4. Find the artifacts section, click on the folder to open it and then click on the file that ends with ".7z"
5. Extract the executable, rename it and drop it into your rpcs3 directory. That way you can have the main version and testing builds  in the same rpcs3 directory.

For linux it's mostly the same apart from step 5 where you just launch the build from different AppImage file.

**Comment by Linear524** [comment]
> You can download artifacts from Cirrus, either by using the bot in our Discord server or by following these steps:
> 
>     1. Scroll to bottom of the page for the sercion about checks, click on "Show all details" if it isn't visible
> 
>     2. Find the task for Windows and click on Details.
> 
>     3. Scroll to bottom and click "View more details on Cirrus CI"
> 
>     4. Find the artifacts section, click on the folder to open it and then click on the file that ends with ".7z"
> 
>     5. Extract the executable, rename it and drop it into your rpcs3 directory. That way you can have the main version and testing builds  in the same rpcs3 directory.
> 
> 
> For linux it's mostly the same apart from step 5 where you just launch the build from different AppImage file.

Thank you (^_^)
( My OS is Win10 @ Xeon E5 2690V4 )

File: rpcs3/Emu/RSX/Common/texture_cache_helpers.h [hunk 1]
```diff
@@ -133,19 +133,15 @@ namespace rsx
 			return gcm_format;
 		}
 
-		static inline u32 get_sized_blit_format(bool is_32_bit, bool depth_format, bool format_conversion)
+		static inline u32 get_sized_blit_format(bool is_32_bit, bool depth_format, bool is_format_convert)
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Common/texture_cache_helpers.h:136]: is_format_convert is unused

**Comment by Crafty-The-Fox** [comment]
fyi dj hero regressed in the new build (and master), looking into it now

**Comment by shockdude** [comment]
Latest PR build works and looks good to me, DJ Hero 1 and 2

**Comment by Linear524** [comment]
Tested GoW3, GoWA, TLoU, Persona5, MGS2, 3, 4, Journey, RDR.
No regression.
Killzone 2 and 3 have broken postprocessing shader, but this is not related to this PR, these bugs related to LLVM update which was fixed after a while on master branch, but not included here...

**Pull request merged**